### PR TITLE
[Ask Mode] Fix execute in interactive commands.

### PR DESCRIPTION
**Pull request by drognanar**
Fixes keybinding and position for `execute in interactive` command.

### CODE REVIEW

**Comment by drognanar** [comment]
@dotnet-bot retest prtest/win/dbg/eta please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn_prtest_win_dbg_eta/4333/
// Retest reason: Restore failed. Issue https://github.com/dotnet/roslyn/issues/9159

**Comment by drognanar** [comment]
retest prtest/win/dbg/eta please

**Comment by drognanar** [comment]
retest prtest/win/dbg/eta please

**Comment by drognanar** [comment]
FYI @dotnet/roslyn-interactive

**Comment by ManishJayaswal** [comment]
@davkean for CR

**Comment by davkean** [comment]
I need some more context on the approach - this doesn't look like what I'd expect.

**Comment by davkean** [comment]
Can you paste a screenshot of the new position?

File: src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct [hunk 1]
```diff
@@ -231,6 +231,17 @@
             -->
 
       <!-- Text editor commands -->
+      <!-- TODO: Remove once https://github.com/dotnet/roslyn/issues/8927 is done. -->
+      <Button guid="guidInteractiveWindowCmdSet" id="cmdidExecuteInInteractiveWindow" priority="0x101" type="Button">
+        <Parent guid="guidInteractiveWindowCmdSet" id="TextEditorGroup" />
+        <CommandFlag>DynamicVisibility</CommandFlag>
+        <CommandFlag>DefaultInvisible</CommandFlag>
+        <Strings>
+          <ButtonText>Execute in Interactive</ButtonText>
```
Comment (davkean) [inline src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct]: How does this get localized?

Comment (davkean) [inline src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct]: And have we missed localization cut-off?

File: src/VisualStudio/Core/Def/ID.InteractiveCommands.cs [hunk 2]
```diff
@@ -11,11 +11,26 @@ internal static class InteractiveCommands
...
+            //public static readonly string CSharpInteractiveCommandSetIdString = "1492DB0A-85A2-4E43-BF0D-CE55B89A8CC6";
+            //public static readonly Guid CSharpInteractiveCommandSetId = new Guid(CSharpInteractiveCommandSetIdString);
 
-            public static readonly string VisualBasicInteractiveCommandSetIdString = "93DF185E-D75B-4FDB-9D47-E90F111971C5";
-            public static readonly Guid VisualBasicInteractiveCommandSetId = new Guid(VisualBasicInteractiveCommandSetIdString);
+            //public static readonly string VisualBasicInteractiveCommandSetIdString = "93DF185E-D75B-4FDB-9D47-E90F111971C5";
+            //public static readonly Guid VisualBasicInteractiveCommandSetId = new Guid(VisualBasicInteractiveCommandSetIdString);
+
+            // TODO: Remove pending https://github.com/dotnet/roslyn/issues/8927 .
+            public const int SmartExecute = 0x0103;
```
Comment (davkean) [inline src/VisualStudio/Core/Def/ID.InteractiveCommands.cs]: Why were these copied in?

File: src/VisualStudio/Core/Def/Implementation/Interactive/CSharpResetInteractiveMenuCommand.cs [hunk 3]
```diff
@@ -23,7 +23,9 @@ internal sealed class CSharpResetInteractiveMenuCommand
 
         protected override CommandID GetResetInteractiveFromProjectCommandID()
         {
-            return new CommandID(ID.InteractiveCommands.CSharpInteractiveCommandSetId, ID.InteractiveCommands.ResetInteractiveFromProject);
+            // TODO: Revert to the code below pending https://github.com/dotnet/roslyn/issues/8927 .
+            // return new CommandID(ID.InteractiveCommands.CSharpInteractiveCommandSetId, ID.InteractiveCommands.ResetInteractiveFromProject);
```
Comment (davkean) [inline src/VisualStudio/Core/Def/Implementation/Interactive/CSharpResetInteractiveMenuCommand.cs]: Not a fan of commenting out code, that's what history is for. Priorities and things changes - if we're shipping with this, we should make the change, and put this commit in the bug.

**Comment by davkean** [comment]
tag @jasonmalinowski

File: src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct (hunk 1 above)
Comment (ManishJayaswal) [inline src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct]: @jinujoseph to comment on loc cut-off?

Comment (Pilchie) [inline src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct]: When I looked at @jinujoseph's mail yesterday, there were still 2 more loc handoffs.

Comment (davkean) [inline src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct]: Yep, agreed, last hand-off 3/11.

Comment (jinujoseph) [inline src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct]: That's right , checkin deadline for last localization prior to escrow is 
​6pm Sunday Mar 6, that would be the build used for last hand off on 3/7 (d14rel)

Comment (drognanar) [inline src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct]: This menu button was already defined in the exact form in previous releases but was removed in an effort to merge the F# and C# execute in interactive command (see #8161).

**Comment by drognanar** [comment]
Updated PR.
Here's a screenshot of the menu command:
execute_in_interactive

**Comment by ManishJayaswal** [comment]
@amcasey @davkean Can you guys review this please?

File: src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct [hunk 4]
```diff
@@ -285,7 +296,9 @@
     <KeyBinding guid="guidInteractiveWindowCmdSet" id="cmdidSearchHistoryPrevious" key1="VK_UP" mod1="Control Alt" editor="guidInteractiveWindow" />
     <KeyBinding guid="guidInteractiveWindowCmdSet" id="cmdidCopyCode" key1="C" mod1="Control Shift" editor="guidInteractiveWindow" />
 
-    <KeyBinding guid="guidVSStd11" id="cmdidExecuteSelectionInInteractive" editor="guidCSharpEditorFactory" mod1="Control" key1="E" mod2="Control" key2="E" />
+    <!-- TODO: Revert to the code below pending https://github.com/dotnet/roslyn/issues/8927 .
+         <KeyBinding guid="guidVSStd11" id="cmdidExecuteSelectionInInteractive" editor="guidCSharpEditorFactory" mod1="Control" key1="E" mod2="Control" key2="E" />-->
+    <KeyBinding guid="guidInteractiveWindowCmdSet" id="cmdidExecuteInInteractiveWindow" editor="guidCSharpEditorFactory" mod1="Control" key1="E" mod2="Control" key2="E" />
```
Comment (amcasey) [inline src/InteractiveWindow/VisualStudio/InteractiveWindow.vsct]: Does this implicitly add Ctrl-E,E as well?  I think it's conventional to have both.

File: src/VisualStudio/Core/Def/ID.InteractiveCommands.cs [hunk 5]
```diff
@@ -16,6 +16,9 @@ internal static class InteractiveCommands
 
             public static readonly string VisualBasicInteractiveCommandSetIdString = "93DF185E-D75B-4FDB-9D47-E90F111971C5";
             public static readonly Guid VisualBasicInteractiveCommandSetId = new Guid(VisualBasicInteractiveCommandSetIdString);
+
+            // TODO: Remove pending https://github.com/dotnet/roslyn/issues/8927 .
+            public const int ExecuteInInteractiveWindow = 0x0010C;
```
Comment (amcasey) [inline src/VisualStudio/Core/Def/ID.InteractiveCommands.cs]: Put this with the other consts?

**Comment by amcasey** [comment]
No objections.

**Comment by MattGertz** [comment]
Approved pending successful completion of test runs.

**Comment by drognanar** [comment]
@dotnet-bot retest this please
// Previous failures:
http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/4453/
http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit64/4429/
// Retest reason: issue #9298

**Comment by drognanar** [comment]
test vsi please

**Comment by drognanar** [comment]
retest prtest/win/vsi/p3 please

**Comment by drognanar** [comment]
retest prtest/win/vsi/p3 please

**Pull request merged**